From 07d34c56e77c5c6454a5e6573634cf28243e5b89 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:55 -0400
Subject: [misc] mei: wd: host_init propagate error codes from called functions

Message-id: <1399041470-21330-23-git-send-email-prarit@redhat.com>
Patchwork-id: 79311
O-Subject: [RHEL7.1 PATCH BZ 1088953 22/77] mei: wd: host_init propagate error codes from called functions
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit 50f67a06713599e18cd9141e413d0e2653bb242b
Author: Tomas Winkler <tomas.winkler@intel.com>
Date:   Mon Oct 21 22:05:39 2013 +0300

    mei: wd: host_init propagate error codes from called functions

    Propagate error codes from called functions, they are correct.

    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/wd.c b/drivers/misc/mei/wd.c
index 8ab1a5a..376f73f 100644
--- a/drivers/misc/mei/wd.c
+++ b/drivers/misc/mei/wd.c
@@ -66,7 +66,7 @@ static void mei_wd_set_start_timeout(struct mei_device *dev, u16 timeout)
 int mei_wd_host_init(struct mei_device *dev)
 {
  struct mei_cl *cl = &dev->wd_cl;
- int i;
+ int id;
  int ret;
 
  mei_cl_init(cl, dev);
@@ -76,19 +76,19 @@ int mei_wd_host_init(struct mei_device *dev)
 
 
  /* check for valid client id */
- i = mei_me_cl_by_uuid(dev, &mei_wd_guid);
- if (i < 0) {
+ id = mei_me_cl_by_uuid(dev, &mei_wd_guid);
+ if (id < 0) {
   dev_info(&dev->pdev->dev, "wd: failed to find the client\n");
-  return -ENOENT;
+  return id;
  }
 
- cl->me_client_id = dev->me_clients[i].client_id;
+ cl->me_client_id = dev->me_clients[id].client_id;
 
  ret = mei_cl_link(cl, MEI_WD_HOST_CLIENT_ID);
 
  if (ret < 0) {
   dev_info(&dev->pdev->dev, "wd: failed link client\n");
-  return -ENOENT;
+  return ret;
  }
 
  cl->state = MEI_FILE_CONNECTING;
-- 
1.7.1